src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -45,6 +45,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     }
 
     /**
+     * @return a unique name this query is identified with
+     */
+    abstract public String queryId();
```

Comment:
There is actually a [standard](http://checkstyle.sourceforge.net/config_modifier.html) for this if you particularly enjoy standards.